manifests/osfamily/suse.pp

```diff
@@ -96,6 +97,13 @@
             value   => $value,
           }
         }
+
+        exec { "refresh-${repo_name}":
```

Comment:
I think the [`ini_setting` resource should have a `before => Exec["refresh-${repo_name}"]`](https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/master/manifests/osfamily/suse.pp#L91-L97) to make sure that the repo's created prior to refreshing it.

Reply:
Could you please point me at a reference to `before`? I can only find it in the context of an `insert_type`, and not as a standalone element. Thanks.

Reply:
https://puppet.com/docs/puppet/5.3/lang_relationships.html#syntax-relationship-metaparameters